### PR TITLE
Refactor atleast_nd

### PR DESCRIPTION
**Pull request by niboshi**
~Supports `numpy.ndarray` in `atleast_<n>d`.~

Also done some refactoring.

### CODE REVIEW

File: cupy/manipulation/dims.py [hunk 1]
```diff
@@ -1,13 +1,49 @@
...
+}
+
+
+def _atleast_nd_helper(n, func_name, arys):
+    """Helper function for atleast_nd functions."""
+
+    res = []
+    for a in arys:
+        if isinstance(a, numpy.ndarray):
+            a = from_data.array(a)
```
Comment (okuta) [inline cupy/manipulation/dims.py]: Please raise error in this case.

Comment (niboshi) [inline cupy/manipulation/dims.py]: Do you mean we shoudn't support numpy.ndarray?

Comment (okuta) [inline cupy/manipulation/dims.py]: Yes, almost CuPy APIs do not provide implicit host to device copy.

File: cupy/manipulation/dims.py [hunk 2]
```diff
@@ -1,13 +1,49 @@
...
+            if a.ndim < n:
+                new_shape = _atleast_nd_shape_map[(n, a.ndim)](a.shape)
+                a = a.reshape(*new_shape)
+        else:
+            raise TypeError('Invalid array type given in {}: {}'.format(
+                func_name, type(a)))
+        res.append(a)
+
+    if len(res) == 1:
+        res = res[0]
```
Comment (okuta) [inline cupy/manipulation/dims.py]: `res, = res` is fast.

File: cupy/manipulation/dims.py (hunk 1 above)
Comment (niboshi) [inline cupy/manipulation/dims.py]: Oh, but how about `ufunc`? It supports `numpy.ndarray` as inputs.

**Comment by niboshi** [comment]
I changed the topic of this PR to simply refactoring.

File: cupy/manipulation/dims.py (hunk 1 above)
Comment (niboshi) [inline cupy/manipulation/dims.py]: I understand that's a misbehavior of `ufunc`. I made an issue: #147

**Comment by okuta** [comment]
LGTM!

**Pull request merged**